app/helpers/application_helper.rb

```diff
@@ -103,6 +103,10 @@ def preload_script_url(url)
     HTML
   end
 
+  def user_path(username)
+    "#{Discourse.base_uri}/u/#{username.downcase}"
+  end
+
```

Comment:
How is this any different to the built-in rails behaviour? Even before this commit, if I do

```
user_path(username)
```
Then it works. The issue only happens when you pass a user object like

```
user_path(User.first)
```


I switched from the `user` to `username` approach in https://github.com/discourse/discourse/commit/cc6d722de1d5f2c56799f02d9f2e4e7333d247da